source/CAS/Client.php

```diff
@@ -997,7 +997,18 @@ public function __construct(
 
         // set to callback mode if PgtIou and PgtId CGI GET parameters are provided
         if ( $this->isProxy() ) {
-            $this->_setCallbackMode(!empty($_GET['pgtIou'])&&!empty($_GET['pgtId']));
+            if(!empty($_GET['pgtIou'])&&!empty($_GET['pgtId'])) {
+                $this->_setCallbackMode(true);
+                $this->_setCallbackModeUsingPost(false);
+            } elseif (!empty($_POST['pgtIou'])&&!empty($_POST['pgtId'])) {
+                $this->_setCallbackMode(true);
+                $this->_setCallbackModeUsingPost(true);
+            } else {
+                $this->_setCallbackMode(false);
+                $this->_setCallbackModeUsingPost(false);
+            }
+
+            
```

Comment:
I would make these use `$_REQUEST` instead?

Reply:
I didn't user _REQUEST as it also contains cookies. The original code cared only about GET, and my use cases was about POST, I didn't want to introduce a third vector by mistake.
In particular there is no point in providing the PGTIOU/PGTID through a cookie, so I felt that it could introduce a vulnerability.